ocaml/alerts/certificate/certificate_check.ml

```diff
@@ -47,33 +47,32 @@ let execute rpc session existing_messages (host, alert) =
       (fun (_, record) -> record.API.message_obj_uuid = host_uuid)
       existing_messages
   in
-  let is_outdated (ref, record) =
-    match alert with
-    | None ->
-        true
-    | Some (body, (name, priority)) ->
-        record.API.message_body <> body
-        || record.API.message_name <> name
-        || record.API.message_priority <> priority
-  in
   (* CA-342551: messages need to be deleted if the pending alert regard the
      same host and has newer, updated information.
      If the pending alert has the same metadata as the existing host message
      it must not be emmited and the existing message must be retained, this
      prevents changing the message UUID.
      In the case there are alerts regarding the host but no alert is pending
      they are not destroyed since no alert is automatically dismissed. *)
-  let outdated_messages = List.filter is_outdated messages_in_host in
-  let replacement_is_pending = List.exists is_outdated messages_in_host in
```

Comment:
Out of interest would simply

```
let replacement_is_pending = List.exists is_outdated messages_in_host || messages_in_host = []
```

have done the trick?

Reply:
It's not exactly the same.

`current` being empty can only happen when `messages_in_host` is empty or `outdated_messages` contains all of its elements.

The equivalent condition would be:
```
let replacement_is_pending = List.for_all is_outdated messages_in_host in
```
But this would mean the removal of elements is behind the flag, which is not the case with the current implementation.

It may not make a difference since only one condition per host should be present. (every time an alert is created the rest are deleted)